modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,21 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev Safe account has two types of signatures: EOA and Smart Contract signatures. While the EOA signature is
+     * fixed in size, the Smart Contract signature can be of arbitrary length. Safe encodes the Smart Contract
+     * signature length in the signature data. If appropriate length checks are not performed during the signature
+     * verification then a malicious bundler can pad additional bytes to the signatures data and make the account pay
+     * more gas than needed for user operation validation and reach the verificationGasLimit.
+     * `_checkSignaturesLength` function checks for the presence of any padded bytes to the `signature` data.
+     * However, there is an edge case that `_checkSignaturesLength` function cannot detect.
+     * Since the `signature` field in UserOp is not part of the UserOp hash a malicious bundler can manipulate the
+     * field(s) storing the signature length and pad additional bytes to the dynamic part of the signatures which will
+     * make `_checkSignaturesLength` to return true. In such cases, it is the responsibility of the signature verifier
+     * (which can be a Safe or any other contract) that supports ERC-1271 and is the owner of the Safe to check for
```

Comment:
It doesn't use ERC-1271 but legacy 1271 in Safe <=1.4.1 and 1271 in Safe >=1.5.0. I would simplify this to:

```suggestion
     * make `_checkSignaturesLength` to return true. In such cases, it is the responsibility of the Safe signature validator
     * implementation, as an account owner, to check for
```

Reply:
https://github.com/safe-global/safe-modules/pull/471/commits/3fad06560a3b9b87cabd95c3af1459e2c39d1cd8